toolkit/tools/imagegen/configuration/packagerepo.go

```diff
@@ -25,15 +25,30 @@ import (
 // repository configuration will be saved in the installed system if specified, and only
 // available during the installation process if not
 type PackageRepo struct {
-	Name    string `json:"Name"`
-	BaseUrl string `json:"BaseUrl"`
-	Install bool   `json:"Install"`
+	Name         string `json:"Name"`
+	BaseUrl      string `json:"BaseUrl"`
+	Install      bool   `json:"Install"`
+	GPGCheck     bool   `json:"GPGCheck"`     // Default value is true
+	RepoGPGCheck bool   `json:"RepoGPGCheck"` // Default value is true
+	GPGKeys      string `json:"GPGKeys"`      // Default value is "file:///etc/pki/rpm-gpg/MICROSOFT-RPM-GPG-KEY file:///etc/pki/rpm-gpg/MICROSOFT-METADATA-GPG-KEY"
```

Comment:
May consider splitting this field into two where one points to rpm gpg check and the other points to repodata gpg check. There might be cases where the users only want to enable one of these two checks instead of both at the same time.

Reply:
I don't see a corresponding field for the .repo files, all the keys just get bundled together for both the metadata and packages. I feel like we should try and stay in line with the behavior from the actual config file here.

As an aside, maybe this should have a key-value entry to support overwriting ANY .repo field...